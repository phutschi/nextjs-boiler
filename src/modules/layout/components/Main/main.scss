*, html {
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif;
}

a {
  color: #000;
  cursor: pointer;
  margin: 0 2px;
  text-decoration: none;

  &:hover {
    background-image: linear-gradient(120deg, #cfd9df 0%, #e2ebf0 100%);
    background-repeat: no-repeat;
    background-size: 100% .1em;
    background-position: 0 103%;
  }

  &:active {
    color: #555;
    background-image: linear-gradient(120deg, #11998e 0%, #11998e 100%);
    background-repeat: no-repeat;
    background-size: 100% .1em;
    transition: all 0.15s ease-in;
  }
}

.main {
  margin: 0;

  &.primary {
    background-color: white;
    color: black;
  }

  &.inverted {
    background-color: black;
    color: white;
  }
}
