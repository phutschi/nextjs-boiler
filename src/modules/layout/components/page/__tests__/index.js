/* global expect */

import * as React from 'react';

import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Page from '../';
import Head from '../../head';
import Main from '../../main';

describe( '<Page />', () => {
    it( 'Snapshot', () => {
        const wrapper = renderer.create( <Page /> ).toJSON();

        expect( wrapper ).toMatchSnapshot();
    } );

    it( 'Renders without props', () => {
        const wrapper = shallow( <Page /> );

        expect( wrapper.find( Head ) ).toHaveLength( 1 );
        expect( wrapper.find( Main ) ).toHaveLength( 1 );
    } );

    it( 'Passes props to Head', () => {
        const title = 'title';
        const subTitle = 'subTitle';
        const wrapper = mount(
            <Page title={title} subTitle={subTitle} />,
        );

        expect( wrapper.find( Head ).props() ).toMatchObject( {
            title,
            subTitle,
        } );
    } );

    it( 'Renders children in Main', () => {
        const title = 'title';
        const subTitle = 'subTitle';
        const wrapper = mount(
            <Page title={title} subTitle={subTitle}>
                <div className="in-main" />
            </Page>
            ,
        );

        expect( wrapper.find( Main ).find( 'div.in-main' ) ).toHaveLength( 1 );
    } );
} );
