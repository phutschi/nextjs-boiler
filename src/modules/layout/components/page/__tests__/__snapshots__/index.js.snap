// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> Snapshot 1`] = `
<div>
  <div
    className="jsx-1439403121 main "
  />
</div>
`;
